frontend/src/modules/notify/plugins/snackbar/index.js

```diff
@@ -0,0 +1,71 @@
+// LICENCE https://github.com/adaptlearning/adapt_authoring/blob/master/LICENSE
+define(function(require) {
+  var _ = require('underscore');
+  var Origin = require('core/origin');
+
+  // are overridden by any values passed to Snackbar
+  var defaults = {
+    type: 'info',
+    text: '',
+    buttonText: '',
+    persist: false,
+    animTime: 400,
+    timeout: 3000,
+    callback: null
+  };
+  var el;
+  var timer;
+  var queue = [];
+
+  var Snackbar = function(data) {
+    if(typeof data === 'string') {
+      data = { text: data };
+    }
+    queue.push(_.extend({},defaults,data));
+    if(queue.length === 1) processQueue();
+  };
+
+  var processQueue = function() {
+    el = document.createElement('div');
+    el.id = 'snackbar';
+    var data = queue[0];
+    var html = ['<div class="body">',data.text,'</div>'];
+    if (data.persist && data.buttonText) {
+      html.push('<button class="close">');
+      html.push(data.buttonText);
+      html.push('</button>');
+    }
```

Comment:
Just to add that these elements can be much easier to create using jQuery:
```js
$el = $('<div>', { id: 'snackbar', 'class': data.type })
  .append($('<div>', { 'class': 'body', text: data.text });
```
or:
```js
$el = $('<div>', {
  id: 'snackbar',
  'class': data.type,
  html: $('<div>', { 'class': 'body', text: data.text })
});
```
Appending the button later would be:
```js
$el.append($('<button>', { 'class': 'close', text: data.buttonText }));
```